openassessment/workflow_batch_update_api.py

```diff
@@ -0,0 +1,595 @@
+            assessment_requirements,
+            course_settings,
+            str(end - start))
+        return workflow
+    except Exception as e:  # pylint: disable=broad-except
+        logger.error(
+            "ORA workflow update for a single submission failed. "
+            "submission_uuid=%s assessment_requirements=%s course_settings=%s. error_message=%s",
+            submission_uuid,
+            assessment_requirements,
+            course_settings,
+            str(e))
+        raise UpdateWorkflowForSubmissionException(str(e)) from e
+
+
+def is_flexible_peer_grading_on(openassessmentblock):
+    """
+    Is flexible peer grading set "ON" for provided ORA block
+
+    Args:
+        openassessmentblock (OpenAssessmentBlock): ORA block
+
+    Returns:
+        bool: True if given ORA is configured with flexible peer grading set "ON"
+    """
+    workflow_requirements = openassessmentblock.workflow_requirements()
+    if workflow_requirements.get('peer') and workflow_requirements['peer'].get('enable_flexible_grading'):
+        return True
+
+    return False
+
+
+def get_blocked_peer_workflows(course_id=None, item_id=None):
+    """
+    Retrieve ORA peer workflows not completed for >7 days
+
+    Returns:
+        list (PeerWorkflow): list of workfows not completed for > 7 days
+    """
+
+    filters = {
+        'created_at__lte': timezone.now() - datetime.timedelta(days=7),
+        'completed_at__isnull': True,
+    }
+    if course_id is not None:
+        filters['course_id'] = course_id
+    if item_id is not None:
+        filters['item_id'] = item_id
+
+    return PeerWorkflow.objects.filter(**filters)
+
+
+def get_workflow_update_parameters(peer_workflows):
+    """
+    Generates dictionary containing data required to update ORA workflows for different scopes.
+    This data structure is used as a local cache to avoid redundant DB queries during
+    batch workflow update process
+
+    Structure:
+    ```
+    {
+      "courses": [
+        {
+          "course_id": "value",
+          "course_settings": {},
+          "assessments": [
+            {
+              "item_id": "value",
+              "assessment_requirements": {},
+              "submissions": [
+                {
+                  "submission_uuid": "value"
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+    ```
+    """
+    workflow_update_parameters_dict = {}
+    store = modulestore()
+    for peer_workflow in peer_workflows:
+        try:
+            # retrieve openassessment block
+            ora_block_key = UsageKey.from_string(peer_workflow.item_id)
+            ora_block = store.get_item(ora_block_key)
+
+            # retrieve course block
+            course_block_key = UsageKey.from_string(peer_workflow.course_id)
+            course_block = store.get_item(course_block_key)
+
+            if is_flexible_peer_grading_on(ora_block) or course_block.force_on_flexible_peer_openassessments:
```

Comment:
This can be simplified, we have a `flexible_peer_grading_enabled` check which takes into account both ORA and course settings.

You may also have to change `course_block.force_on_flexible_peer_openassessments` to a `course_block.get('force_on_flexible_peer_openassessments')` if there's ever a case where the course block doesn't have that setting yet.

Reply:
`is_flexible_peer_grading_on(ora_block)` needs to extract some data from ora_block, that's why it makes sense to wrap it with separate method. course_block.force_on_flexible_peer_openassessments returns `boolean` no need to move it

Reply:
What I'm suggesting is to leverage the `flexible_peer_grading_enabled` check instead, which will already take both block and course level settings into account and lets us define that logic in fewer places.

Reply:
I have made update for this